### ISSUE TITLE
Properties defined in portrait: blocks don't inherit properly

### ISSUE THREAD

**Issue by ConradIrwin** [colinta/teacup#30]
I tried to do this:

``` ruby
style :foo,
  portrait: {top: 120}

style :bar, extends: :foo,
  portrait: {left: 100}
```

Unfortunately the way portrait is currently implemented means that the "top:" gets ignored.

**Comment by ConradIrwin** [comment]
Hmm. I notice this problem also applies to `:layout`; but I think the fix in that case is different (because we don't need to combine the properties in :layout with the top-level).

Also the implementation of `shouldRotateToOrientation` should probably check _all_ classes, not just the top-level one.

**Comment by colinta** [comment]
Yeah, I thought of that at one point, that the merging strategy doesn't merge portrait, landscape, and layer properly.

So then it's a question of _which_ hashes to merge, which I'm not a big fan of, I'd rather have it be one consistent thing... but then again, orientations are their own thing.

I've pushed, to v0.2.0 (759b494), a patch that merges only the orientation hashes.  If you go back a few commits, you'll see code that merges _all_ hashes.

As I'm pushing this, though, I'm worried about a bigger issue...

**Comment by colinta** [comment]
There is still the issue that we are not resetting the values that we set in `UIView#style`, though I guess this only matters when you are changing orientation or changing stylesheets, and the other orientation / stylesheet doesn't overwrite you changes.  Meh, not as a big deal as I thought earlier.

**Comment by colinta** [comment]
yup, this is fixed, too, in v0.3.0 :-)
